lib/puppet/type/file.rb

```diff
@@ -668,6 +668,7 @@ def perform_recursion(path)
       :links => self[:links],
       :recurse => (self[:recurse] == :remote ? true : self[:recurse]),
       :recurselimit => self[:recurselimit],
+      :source_permissions => self[:source_permissions],
```

Comment:
@joshcooper as far as I could tell, it was working previously because we only ignored source permissions if the option was explicitly set to `:ignore`, and since we weren't actually passing any options in, `source_permissions` was always `nil` (except when we actually were passing `:ignore` in).

`@source_permissions_ignore = source_permissions == :ignore      # resulted in :use when source_permissions was nil`

The change that caused the regression essentially broke this weird assumption that `nil` should use source permissions by saying that if `source_permissions` is `nil`, then we should ignore source permissions. This is why we now need the agent to specify `:use` if it actually needs to use source permissions. Anything that was depending on simply not setting `source_permissions` to `:use` them would be broken. I believe that's just external facts, however.

`@source_permissions_ignore = (!source_permissions || source_permissions == :ignore)`